include/ensmallen_bits/de/de.hpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file de.hpp
+ * @author Rahul Ganesh Prabhu
+ *
+ * Differential Evolution
+ * An evolutionary algorithm used for global optimization of arbitrary functions
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <tuple>
+#include <vector>
+
+#ifndef ENSMALLEN_DE_DE_HPP
+#define ENSMALLEN_DE_DE_HPP
+
+namespace ens {
+
+/**
+ * Differential evolution is a stochastic evolutionary algorithm used for global optimization. 
```

Comment:
Do you think you could cite the "Differential Evolution - a Simple and Efficient Adaptive Scheme for Global Optimization over Continuous Spaces" paper; example: https://github.com/mlpack/ensmallen/blob/7ecc62ed1d7976ec55d844e10dca6686fab5de55/include/ensmallen_bits/padam/padam.hpp#L27-L36